### ISSUE TITLE
[feature request] PPO compute_action deterministic

### ISSUE THREAD

**Issue by Guilherme-B** [ray-project/ray#4496]
I am requesting the ability to grab a PPO agent trained for a stochastic distribution. and add the capability to compute_actions in a deterministic manner, as[ pointed here](https://groups.google.com/forum/#!topic/ray-dev/MxlguIGfkho).
This feature will allow using rollouts to compare agent behavior on small scales, in a comparable and contained environment, to see to what extent has the agent learnt, without the action's randomness.

**Comment by FedericoFontana** [comment]
One workaround for discrete action spaces is to pick the action corresponding to the argmax of the logits.

```
policy = agent.get_policy()
action, _, info = policy.compute_single_action(observation)
logits = info['logits']
best_action_id = logits.argmax()  # deterministic
```

**Comment by Guilherme-B** [comment]
I did consider doing that, and I'll probably do it, problem is, its only for discrete action spaces - it is a solution for now nonetheless.

Thanks for your help @FedericoFontana

**Comment by Guilherme-B** [comment]
Ran some preliminary tests. Seems to be working, although something rather odd is happening:
- For most observations, the agent selects the exact same action. This is likely to be an indicator of the agent not being to long-term plan (perhaps due to a small amount of training iterations, 500 for an environment of 3.000 possible actions and a MultiDiscrete(250,11) environment - where 11 is disaster).

**Comment by ericl** [comment]
btw for continuous spaces, you can set only the second half of the logits to zero in a custom model (those happen to parameterize the standard deviation).

**Comment by Guilherme-B** [comment]
Sorry for the long delay, took me a while to get AWS to approve the new instances. Ran tests for a few more training iterations - about 5.5k - and the result is rather odd.

I recorded the actions under custom metrics, to see their evolution throughout time, so I can plot them and get some understanding of what it is learning.

Here is the chart (sorry for the ugly thing, just using excel to make it faster):
![image](https://user-images.githubusercontent.com/38859748/56098118-09726080-5ef5-11e9-9233-54804b711502.png)

Despite the average of rewards being around 110, what the agent selects on rollout is always in the 240-260 range (the maximum of the selected actions during that period). Is this expected? Can it simply be caused by high entropy factors since it is early in training (I guess 5.5k is not enough for PPO to learn)?

**Comment by rw-anderson** [comment]
I tried to use this workaround for lack of a deterministic mode in the policy eval:

policy = agent.get_policy()
action, _, info = policy.compute_single_action(observation)
logits = info['logits']
best_action_id = logits.argmax()  # deterministic

but found that there is no "logits" key in the info dictionary. Is this now called "action_dist_inputs"?
